### PR TITLE
added functionality to view tags

### PR DESCRIPTION
**Pull request by aliparadise**
#44 View All Tags

Added file Tag.cs to Models directory. 

Added file ITagRepository.cs and TagRepository.cs to Repository directory. 

Added file TagController.cs to Controller Repository.

Added ITagRepository/TagRepository to Startup.cs.

Added file tagManager.js to modules directory.

Added Tag.js and TagList.js to components directory.

Added a tag route path to ApplicationViews.js

#### Steps to Review

1. Checkout this branch locally.
   ```
   git fetch --all
   git checkout ap-viewTags
   ```
2. Open a new Terminal tab (⌘T) and navigate to the server directory.
3. Test app functionality.
   > When a user logs in and is authenticated, the user should be able to click on the "Tags Management" link in the navbar. The 
       user should be directed to a list of tags in alphabetical order.

### CODE REVIEW

**Review by sorachung — APPROVED** [review]
Functionality works great. Code looks great. Good job!

File: Tabloid/Controllers/TagController.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Tabloid.Repositories;
+
+namespace Tabloid.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+
+    public class TagController : ControllerBase
```
Comment (heymonicakay) [inline Tabloid/Controllers/TagController.cs:10]: Should this controller be auth protected? do we need any other auth checks on the route?

File: Tabloid/client/src/components/ApplicationViews.js [hunk 2]
```diff
@@ -13,6 +14,10 @@ export default function ApplicationViews({ isLoggedIn }) {
           {isLoggedIn ? <Hello /> : <Redirect to="/login" />}
         </Route>
 
+        <Route path="/tag">
```
Comment (heymonicakay) [inline Tabloid/client/src/components/ApplicationViews.js:17]: should this be an exact path?

File: Tabloid/client/src/components/TagList.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React, { useEffect, useState } from "react";
+import Tag from "./Tag";
+import { getAllTags } from "../modules/tagManager";
+
+export default function TagList() {
+  const [ tags, setTags ] = useState([]);
+
+  useEffect(() => {
+    getAllTags().then(setTags);
```
Comment (heymonicakay) [inline Tabloid/client/src/components/TagList.js:9]: Consider creating a function that gets and sets tags, and calling that function in this useEffect instead. That function can be used again when you need to update state after editing or deleting a tag later on.

File: Tabloid/client/src/modules/tagManager.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { getToken } from "./authManager";
+
+const baseUrl = "/api/tag";
+
+export const getAllTags = () => {
+  return getToken().then((token) => {
+    return fetch(baseUrl, {
+          method: "GET",
+          headers: {
+              Authorization: `Bearer ${token}`
```
Comment (heymonicakay) [inline Tabloid/client/src/modules/tagManager.js:10]: Great job passing in the token to authorize this request! just make sure the API is looking for this authorization on the route for getting all tags.

**Review by heymonicakay — CHANGES_REQUESTED** [review]

**Comment by heymonicakay** [comment]
Which issues are related to this PR?

File: Tabloid/Controllers/TagController.cs (hunk 1 above)
Comment (aliparadise) [reply Tabloid/Controllers/TagController.cs:10]: That's a good question. In my mind I read this section of the issue ticket "NOTE: For the time being it is acceptable to treat all users as admin users. There is a future story about enforcing user permissions" to mean that we would enforce the auth checks on a later ticket?

File: Tabloid/client/src/components/TagList.js (hunk 3 above)
Comment (aliparadise) [reply Tabloid/client/src/components/TagList.js:9]: good idea!

File: Tabloid/client/src/components/ApplicationViews.js (hunk 2 above)
Comment (aliparadise) [reply Tabloid/client/src/components/ApplicationViews.js:17]: Yes, I will make that change.

File: Tabloid/Controllers/TagController.cs (hunk 1 above)
Comment (heymonicakay) [reply Tabloid/Controllers/TagController.cs:10]: @aliparadise Good question! So you don't need to check for authorization (ie the user doesn't need to be an admin), but the user should be authenticated (ie the user should be logged in) in order to see the tags. Let me know if you have questions about the difference there!

Comment (aliparadise) [reply Tabloid/Controllers/TagController.cs:10]: I went back over that chapter and realized I was confusing myself with the use of "auth" for authenticated vs authorized. Now it's coming back to me!

Comment (aliparadise) [reply Tabloid/Controllers/TagController.cs:12]: Changes made.

File: Tabloid/client/src/components/ApplicationViews.js (hunk 2 above)
Comment (aliparadise) [reply Tabloid/client/src/components/ApplicationViews.js]: Changes made

File: Tabloid/client/src/components/TagList.js (hunk 3 above)
Comment (aliparadise) [reply Tabloid/client/src/components/TagList.js]: Changes made.

**Review by heymonicakay — APPROVED** [review]

**Pull request merged**